ext/eft.py

```diff
@@ -0,0 +1,55 @@
+'''
+Cog that copies EFT-formatted fits into an embed.
+'''
+
+import logging
+import urllib.request, json, re
+
+from datetime import datetime
+
+import discord.ext.commands as commands
+import utils.checks as checks
+from discord.embeds import Embed
+
+def setup(bot):
+    'Adds the cog to the provided Discord bot'
+    bot.add_cog(EFT(bot))
+    
+class EFT:
+    'A cog that copies EFT-formatted fits to an embed'
+    def __init__(self, bot):
+        self.logger = logging.getLogger(__name__)
+        self.bot = bot
+    
+    async def on_message(self, message):
+        content = str(message.content)
+        p = re.compile('\[(.*?)\]\n(.*)', re.DOTALL)
+        m = p.match(content)
+        if(m != None):
+            embed = self.eftembed(m.group(1),m.group(2))
+            await self.bot.send_message(message.channel, embed = embed)
+            await self.bot.delete_message(message)
+            
+    def eftembed(self, title, fit):
+        embed = Embed()
+        
+        p = re.compile('(.*),.*')
+        m = p.match(title)
+        type = m.group(1)
+        type = type.replace(" ", "%20")
+        
+        file_url = 'https://www.fuzzwork.co.uk/api/typeid.php?typename=' + type
+        
+        with urllib.request.urlopen(file_url) as url:

```

Comment:
Don't use urllib.request, the bot has an asynchronous http client already associated with it. Look in killmails.py for reference.